nautobot_device_lifecycle_mgmt/models.py

```diff
@@ -631,3 +632,141 @@ def to_csv(self):
+    published_date = models.DateField(verbose_name="Published Date")
+    link = models.URLField()
+    status = StatusField(
+        null=True,
+        blank=True,
+        on_delete=models.PROTECT,
+        to="extras.status",
+    )
+    description = models.CharField(max_length=255, blank=True, null=True)
+    severity = models.CharField(max_length=50, default=choices.CVESeverityChoices.NONE)
+    cvss = models.FloatField(blank=True, null=True, verbose_name="CVSS Base Score")
+    cvss_v2 = models.FloatField(blank=True, null=True, verbose_name="CVSSv2 Score")
+    cvss_v3 = models.FloatField(blank=True, null=True, verbose_name="CVSSv3 Score")
+    fix = models.CharField(max_length=255, blank=True, null=True)
+    comments = models.TextField(blank=True)
+
+    csv_headers = [
+        "name",
+        "published_date",
+        "link",
+        "status",
+        "description",
+        "severity",
+        "cvss",
+        "cvss_v2",
+        "cvss_v3",
+        "fix",
+        "comments",
+    ]
+
+    class Meta:
+        """Meta attributes for the class."""
+
+        verbose_name = "CVE"
+
+        ordering = ("severity", "name")
+
+    def get_absolute_url(self):
+        """Returns the Detail view for CVELCM models."""
+        return reverse("plugins:nautobot_device_lifecycle_mgmt:cvelcm", kwargs={"pk": self.pk})
+
+    def __str__(self):
+        """String representation of the model."""
+        return f"{self.name}"
+
+    def to_csv(self):
+        """Return fields for bulk view."""
+        return (
+            self.name,
+            self.published_date,
+            self.link,
+            self.status,
+            self.description,
+            self.severity,
+            self.cvss,
+            self.cvss_v2,
+            self.cvss_v3,
+            self.fix,
+            self.comments,
+        )
+
+
+@extras_features(
+    "custom_fields",
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "relationships",
+    "webhooks",
+    "statuses",
+)
+class VulnerabilityLCM(PrimaryModel):
+    """VulnerabilityLCM is a model representation of vulnerability that affects a device."""
+
+    cve = models.ForeignKey(CVELCM, on_delete=models.CASCADE, blank=True, null=True)
+    software = models.ForeignKey(SoftwareLCM, on_delete=models.CASCADE, blank=True, null=True)
+    device = models.ForeignKey(Device, on_delete=models.CASCADE, blank=True, null=True)
+    inventory_item = models.ForeignKey(InventoryItem, on_delete=models.CASCADE, blank=True, null=True)
+    status = StatusField(
+        null=True,
+        blank=True,
+        on_delete=models.PROTECT,
+        to="extras.status",
+    )
+
+    csv_headers = [
+        "cve",
+        "software",
+        "device",
+        "inventory_item",
+        "status",
+    ]
+
+    class Meta:
+        """Meta attributes for the class."""
+
+        verbose_name = "Vulnerability"
+        verbose_name_plural = "Vulnerabilities"
```

Comment:
I see your point, but even without setting the plural name the endpoints don't match. The API `GET` endpoint will still be the singular `/vulnerability/` to get a list of all (or filtered) objects and to get a single object by ID would be `/vulnerability/{id}/`. The GraphQL queries would be `vulnerabilitys`/`vulnerabilities` for all/filtered and `vulnerability` for a single object.

Reply:
Overriding `verbose_name_plural` to be more grammatically correct should be perfectly fine - numerous core models do so. If this were an existing model, *changing* it would be a breaking change to the GraphQL schema and so we'd want to avoid that, but since this is a new model it should be fine.